File: apps/tv/tvos/MaktabaTV/Models/SearchResult.swift
import Foundation

/// A single hit from `GET /api/search?q=…`. The TV search screen
/// renders these as a poster grid identical to the library grid, so
/// the shape deliberately overlaps with `Media`.
struct SearchResult: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    var kind: String?
    var posterURL: String?
    var libraryID: String?

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case kind
        case posterURL = "poster_url"
        case libraryID = "library_id"
    }

    /// Adapts a search hit into the grid's `Media` model.
    var asMedia: Media {
        Media(id: id, title: title, posterURL: posterURL, durationSec: nil, hlsURL: nil)
    }
}

/// Envelope for the search endpoint.
struct SearchResponse: Codable {
    let items: [SearchResult]
}

/// A single autocomplete suggestion from `GET /api/search/suggest`.
struct SearchSuggestion: Codable, Identifiable, Hashable {
    var id: String { text }
    let text: String
}

File: apps/tv/android/app/src/main/java/com/hamzalabs/maktaba/tv/data/models/SearchResult.kt
package com.hamzalabs.maktaba.tv.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** A single search hit; shaped to overlap with [Media] for the grid. */
@Serializable
data class SearchResult(
    val id: String,
    val title: String,
    val kind: String? = null,
    @SerialName("poster_url") val posterUrl: String? = null,
    @SerialName("library_id") val libraryId: String? = null,
) {
    fun asMedia() = Media(id = id, title = title, posterUrl = posterUrl)
}

/** Envelope for `GET /api/search` and `GET /api/libraries/{id}/items`. */
@Serializable
data class SearchResponse(
    val items: List<SearchResult> = emptyList(),
)
